azext_edge/edge/providers/check/dataflow.py

```diff
@@ -342,16 +483,87 @@ def _process_dataflow_destinationsettings(
 
 
 def _process_endpoint_authentication(
-    endpoint_settings: dict, check_manager: CheckManager, target: str, namespace: str, padding: int
+    endpoint_settings: dict, check_manager: CheckManager, target: str, namespace: str, padding: int, detail_level: int
 ) -> None:
-    auth = endpoint_settings.get("authentication", {})
+    auth_property_dict = {
+        DataFlowEndpointAuthenticationType.access_token.value: {
+            "key": "accessTokenSettings",
+            "displays": [
+                ("Secret Reference", "secretRef"),
+            ],
+        },
+        DataFlowEndpointAuthenticationType.system_assigned.value: {
+            "key": "systemAssignedManagedIdentitySettings",
+            "displays": [
+                ("Audience", "audience"),
+            ],
+        },
+        DataFlowEndpointAuthenticationType.user_assigned.value: {
+            "key": "userAssignedManagedIdentitySettings",
+            "displays": [
+                ("Client ID", "clientId"),
+                ("Scope", "scope"),
+                ("Tenant ID", "tenantId"),
+            ],
+        },
+        DataFlowEndpointAuthenticationType.x509.value: {
+            "key": "x509CertificateSettings",
+            "displays": [
+                ("Secret Ref", "secretRef"),
+            ],
+        },
+        DataFlowEndpointAuthenticationType.service_account_token.value: {
+            "key": "serviceAccountTokenSettings",
+            "displays": [
+                ("Audience", "audience"),
+            ],
+        },
+        DataFlowEndpointAuthenticationType.sasl.value: {
+            "key": "saslSettings",
+            "displays": [
+                ("Type", "saslType"),
+                ("Secret Ref", "secretRef"),
+            ],
+        },
+        DataFlowEndpointAuthenticationType.anonymous.value: {
+            "key": "anonymousSettings",
+            "displays": [],
+        },
+    }
+
+    auth: dict = endpoint_settings.get("authentication", {})
     auth_method = auth.get("method")
+
+    # display unkown auth method
+    if auth_method not in auth_property_dict:
+        check_manager.add_display(
+            target_name=target,
+            namespace=namespace,
+            display=Padding(f"[red]Unknown authentication method: {auth_method}", (0, 0, 0, padding)),
+        )
+        return
+
+    # display auth method
     check_manager.add_display(
         target_name=target,
         namespace=namespace,
         display=basic_property_display(label="Authentication Method", value=auth_method, padding=padding),
     )
-    # TODO - add displays for various auth types (refs, identity names, etc)
+
+    # show details for various auth methods
```

Comment:
want to double check, I see a secretRef property, if that's the Kubernetes secret, should we add a check for that?

Reply:
We should eventually - there are a few other refs in this set of checks (schemaref, secretref, persistentvolumeclaimref, etc) that likely need validation - I can address those in a future PR that adds a way for us to verify various "refs" exist

Reply:
sounds good